### PR TITLE
Start fixing sorting on API

### PR DESCRIPTION
**Pull request by joecohens**
/cc @CachetHQ/owners what do you think of something in this?

Fixes https://github.com/CachetHQ/Cachet/issues/1489

### CODE REVIEW

File: app/Models/Incident.php [hunk 1]
```diff
@@ -63,6 +63,35 @@ class Incident extends Model implements HasPresenter
...
+        'name',
+        'status',
+        'visible',
+        'message',
+    ];
+
+    /**
+     * Adds a sort scope.
+     *
+     * @param \Illuminate\Database\Eloquent\Builder $query
```
Comment (GrahamCampbell) [inline app/Models/Incident.php]: missing params

File: app/Models/Incident.php [hunk 2]
```diff
@@ -63,6 +63,35 @@ class Incident extends Model implements HasPresenter
...
+     *
+     * @return \Illuminate\Database\Eloquent\Builder
+     */
+    public function scopeSort($query, $column, $direction)
```
Comment (GrahamCampbell) [inline app/Models/Incident.php]: missing typehints

**Comment by jbrooksuk** [comment]
Nice! This provides a much encapsulated system. I like it :)

**Comment by jbrooksuk** [comment]
Fixed the typehints.

**Comment by jbrooksuk** [comment]
Done :)

**Comment by GrahamCampbell** [comment]
Is this done then?

**Comment by jbrooksuk** [comment]
Yup.

**Comment by GrahamCampbell** [comment]
:ship:

**Comment by jbrooksuk** [comment]
:boom:

**Comment by joecohens** [comment]
:tada:

**Pull request merged**